src/tracks/InstrumentTrack.cpp

```diff
@@ -815,33 +816,44 @@ void InstrumentTrack::loadTrackSpecificSettings( const QDomElement & thisElement
 			{
 				m_audioPort.effects()->restoreState( node.toElement() );
 			}
-			else if( node.nodeName() == "instrument" )
+			else if(node.nodeName() == "instrument")
 			{
 				typedef Plugin::Descriptor::SubPluginFeatures::Key PluginKey;
-				PluginKey key( node.toElement().elementsByTagName( "key" ).item( 0 ).toElement() );
-
-				delete m_instrument;
-				m_instrument = NULL;
-				m_instrument = Instrument::instantiate(
-					node.toElement().attribute( "name" ), this, &key);
-				m_instrument->restoreState( node.firstChildElement() );
+				PluginKey key(node.toElement().elementsByTagName("key").item(0).toElement());
 
-				emit instrumentChanged();
+				// don't delete vestige instrument in preview mode because loading vst
+				// can take long time, so better try change settings without recreating
+				// everything (this speeds up the loading of subsequent presets from one plugin)
+				if (m_previewMode && m_instrument && m_instrument->nodeName() == "vestige" &&
+						node.toElement().attribute("name") == "vestige")
```

Comment:
I think you can compare two names directly rather than comparing it to `"vestige"` because this will help saving time for other instruments as well. I think we don't need to re-instantiate instrument plugins in general cases unless something errored on the previous load.

Reply:
Yeah, that can be changed if we want save time for other instruments.